crypto/lhash/lhash.c

```diff
@@ -133,20 +144,26 @@ void *OPENSSL_LH_delete(OPENSSL_LHASH *lh, const void *data)
 
 void *OPENSSL_LH_retrieve(OPENSSL_LHASH *lh, const void *data)
 {
+    struct lh_node_stats st;
     unsigned long hash;
     OPENSSL_LH_NODE **rn;
-    void *ret;
-    int scratch;
+    void *ret = NULL;
+    int miss = 0;
 
     lh->error = 0;
-    rn = getrn(lh, data, &hash);
+    rn = getrn_stats(lh, data, &hash, &st);
 
-    if (*rn == NULL) {
-        CRYPTO_atomic_add(&lh->num_retrieve_miss, 1, &scratch, lh->retrieve_stats_lock);
-        return NULL;
-    } else {
+    if (*rn == NULL)
+        miss = 1;
+    else
         ret = (*rn)->data;
-        CRYPTO_atomic_add(&lh->num_retrieve, 1, &scratch, lh->retrieve_stats_lock);
+
+    if (lh->stats_lock != NULL) {
+        CRYPTO_THREAD_write_lock(lh->stats_lock);
+        lh->num_retrieve_miss += miss;
```

Comment:
unh maybe put those in a "if miss" statement?

Reply:
:(

An attempt to provide constant overhead.

I'll switch to a conditional statement instead.